python-sdk/src/astro/databricks/databricks_api_utils.py

```diff
@@ -0,0 +1,144 @@
+import fileinput
+import pathlib
+import time
+from pathlib import Path
+from typing import Dict
+
+from attrs import define, field
+from databricks_cli.dbfs.api import DbfsApi, DbfsPath
+from databricks_cli.jobs.api import JobsApi
+from databricks_cli.runs.api import RunsApi
+from databricks_cli.sdk.api_client import ApiClient
+
+from astro.databricks.load_file.load_file_python_code_generator import render
+
+cwd = pathlib.Path(__file__).parent
+
+
+@define
+class DeltaOption:
+    """
+    Helper class for rendering options into COPY_INTO Databricks SQL statements
+    """
+
+    option_dict: dict = field(factory=dict)
+
+    def empty(self):
+        return len(self.option_dict or {}) == 0
+
+    def convert_to_option_string(self):
+        return ",".join([f"\\'{k}\\' = \\'{v}\\'" for k, v in self.option_dict.items()])
+
+
+def generate_file(
+    data_source_path: str,
+    table_name: str,
+    source_type: str,
+    load_options: Dict[str, str],
+    output_file_path: Path,
+    copy_options: DeltaOption,
+    format_options: DeltaOption,
+    file_type: str = "",
+):
+    """
+    In order to run autoloader jobs in databricks, we need to generate a python file that creates a pyspark job.
+    This function uses jinja templating to generate a file with all necessary user inputs
+    :param format_options: FORMAT_OPTIONS options to pass to a COPY INTO command
+    :param copy_options: COPY_OPTIONS options to pass into a COPY INTO command
+    :param data_source_path: The path where the data we're injesting lives (e.g. the s3 bucket URL)
+    :param table_name: The delta table we would like to push the data into
+    :param source_type: the source location (e.g. s3)
+    :param load_options: any additional load options the user would like to inject into their job
+    :param output_file_path: where the generated file should be placed.
+    :param file_type: when using the COPY INTO command, Spark requires explicitly stating the data type you are loading.
+    For individual files we can infer the file type, but if you are loading a directory please explicitly state
+    the file type in the File object.
+    :return: output file path
+    """
+    render(
+        Path("jinja_templates/load_file_to_delta.py.jinja2"),
+        {
+            "source_type": source_type,
+            "table_name": table_name,
+            "load_dict": load_options,
+            "input_path": data_source_path,
+            "file_type": file_type.upper(),
+            "copy_options": copy_options,
+            "format_options": format_options,
+        },
+        output_file_path,
+    )
+
+    # Since we had to use jinja2 safe strings, we want to inline replace \' with '
+    # This is not a security concern now that jinja2 is no longer running.
```

Comment:
We don't, jinja does. Jinja doesn't like quotes because it offers an opportunity to do escapes (this is concerning since jinja can run arbitrary code). Because of this it automatically adds a \ to make the quotations "safe".